manager/src/main/java/com/redhat/service/smartevents/manager/BridgesServiceImpl.java

```diff
@@ -90,6 +91,53 @@ public Bridge createBridge(String customerId, String organisationId, String owne
         return bridge;
     }
 
+    @Override
+    @Transactional
+    public Bridge updateBridge(String bridgeId, String customerId, BridgeRequest bridgeRequest) {
+        // Extract existing definition
+        Bridge existingBridge = getBridge(bridgeId, customerId);
+
+        if (!existingBridge.isActionable()) {
+            throw new BridgeLifecycleException(String.format("Bridge with id '%s' for customer '%s' is not in an actionable state.",
+                    bridgeId,
+                    customerId));
+        }
+
+        BridgeDefinition existingDefinition = existingBridge.getDefinition();
+
+        // Validate update.
+        // Name cannot be updated.
+        if (!Objects.equals(existingBridge.getName(), bridgeRequest.getName())) {
+            throw new BadRequestException("It is not possible to update the Bridge's name.");
+        }
+
+        // Construct updated definition
+        Action updatedErrorHandler = bridgeRequest.getErrorHandler();
+        BridgeDefinition updatedDefinition = new BridgeDefinition(Objects.nonNull(updatedErrorHandler) ? updatedErrorHandler : null);
```

Comment:
`Objects.nonNull(updatedErrorHandler) ? updatedErrorHandler : null`
AFAIK that is the same as using `updatedErrorHandler` directly, right? Or is there some specific meaning for this construct?

Reply:
hahaha, yes it is.. IDK what I was thinking. Changed.